src/impl_constructors.rs

```diff
@@ -32,17 +32,25 @@ impl<S, A> ArrayBase<S, Ix1>
 {
     /// Create a one-dimensional array from a vector (no copying needed).
     ///
+    /// **Panics** if the length is greater than `isize::MAX`.
+    ///
     /// ```rust
     /// use ndarray::Array;
     ///
     /// let array = Array::from_vec(vec![1., 2., 3., 4.]);
     /// ```
     pub fn from_vec(v: Vec<A>) -> Self {
+        assert!(
+            v.len() <= ::std::isize::MAX as usize,
+            "Length must fit in `isize`.",
+        );
```

Comment:
Same size_of A == 0 check

Reply:
Good catch. This case doesn't even require the extra assumption.